chore: release 2.3.0 (#1382)
kumahq/kuma-website: app/_posts/2023-06-23-kuma-2-3-0.md

---
title: Kuma 2.3 release with MeshTCPRoute, GAMMA support and much more
description: Kuma 2.3 release with MeshTCPRoute, GAMMA support and much more
date: 2023-06-23
headerImage: /assets/images/blog/test-header.jpg
tags:
  - Release
---

We’re excited to announce the release of Kuma 2.3. This new minor release gets us one step closer to releasing targetRef policies as defaults. 

In order to take advantage of the latest and greatest in service mesh, we strongly suggest upgrading to Kuma 2.3. Upgrading is easy through `kumactl` or Helm.

## Notable features:

* 🚀 targetRef policies are now working with `MeshGateway`.
* 🚀 `MeshTCPRoute` which enables to control traffic routing for TCP services.
* 🚀 Helm improvements with more secure defaults. 
* 🚀 Possibility to skip creating default policies when creating a mesh. 
* 🚀 Performance improvements on Kubernetes. 

> * 🚀 Add hosts selection predicates to `MeshRetry`

Reviewer: I'm not sure if items in this list should end with a dot, but we should be consistent.
```suggestion
* 🚀 Add hosts selection predicates to `MeshRetry`.
```

* 🚀 Initial GAMMA routes support. 

> * 🚀 Continuous improvements to the look and feel of the GUI 

Reviewer: I'm not sure if items in this list should end with a dot, but we should be consistent.
```suggestion
* 🚀 Continuous improvements to the look and feel of the GUI.
```

> * 🚀 Continuous improvements to the look and feel of the GUI 

Reviewer: I'm not sure if items in this list should end with a dot, but we should be consistent.
```suggestion
* 🚀 Continuous improvements to the look and feel of the GUI.
```

> * 🚀 Upgrade to Envoy 1.26

Reviewer: I'm not sure if items in this list should end with a dot, but we should be consistent.
```suggestion
* 🚀 Upgrade to Envoy 1.26.
```

> And a lot more! Checkout the full [release notes](https://github.com/kumahq/kuma/releases/tag/2.3.0) to see everything in this release.

Reviewer: Shouldn't be `check out` instead of `checkout`?
```suggestion
And a lot more! Check out the full [release notes](https://github.com/kumahq/kuma/releases/tag/2.3.0) to see everything in this release.
```

### `MeshTCPRoute`

> In 2.2.0 we release [MeshHTTPRoute](https://kuma.io/docs/2.2.x/policies/meshhttproute/) which enabled traffic shaping and shifting of HTTP requests.

Reviewer: ```suggestion
In 2.2.0 we released [MeshHTTPRoute](https://kuma.io/docs/2.2.x/policies/meshhttproute/) which enabled traffic shaping and shifting of HTTP requests.
```

In this release we're adding the equivalent [`MeshTCPRoute`](https://kuma.io/docs/2.3.x/policies/meshtcproute/) for TCP traffic.

The usage is very similar to other targetRef policies. For example you can here do a progressive traffic routing of a service from one version to another.

```yaml
apiVersion: kuma.io/v1alpha1
kind: MeshTCPRoute
metadata:
  name: tcp-route-1
  namespace: kuma-system
  labels:
    kuma.io/mesh: default
spec:
  targetRef:
    kind: MeshService
    name: frontend_kuma-demo_svc_8080
  to:
    - targetRef:
        kind: MeshService
        name: backend_kuma-demo_svc_3001
      rules:
        - default:
            backendRefs:
              - kind: MeshServiceSubset
                name: backend_kuma-demo_svc_3001
                tags:
                  version: "1.0"
                weight: 90
              - kind: MeshServiceSubset
                name: backend_kuma-demo_svc_3001
                tags:
                  version: "2.0"
                weight: 10

> 

Reviewer: ```suggestion
```

```

Note that cross zone routing hasn't been implemented yet. This will be done in the next release ([#6989](https://github.com/kumahq/kuma/issues/6989)).

### `MeshGateway` support in new policy

Up until now it was complicated to use the `targetRef` policies with the builtin gateway (read [#6560](https://github.com/kumahq/kuma/issues/6560) to understand why).

> In the coming release this will be made even simpler with a new targetRef type `MeshGateway` which will enable you to identify a gateway by its name.

Reviewer: ```suggestion
In the upcoming release this will be made even simpler with a new targetRef type `MeshGateway` which will enable you to identify a gateway by its name.
```

You can check [MADR 027](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/027-meshgateway-targetref.md) for a teaser.

### Initial GAMMA support

> [GAMMA](https://gateway-api.sigs.k8s.io/contributing/gamma/) is an initiative within the GatewayAPI community to create common interfaces for multiple service meshes.

Reviewer: I'm not sure, but the word `multiple` doesn't sound right to me here. I might be wrong though.
```suggestion
[GAMMA](https://gateway-api.sigs.k8s.io/contributing/gamma/) is an initiative within the GatewayAPI community to create common interfaces for service meshes.
```

We believe this could ease the integration of service meshes with the rest of the Kubernetes ecosystem.
This is the primary reason why Kuma has been active in this effort from the start.

> In 2.3.x we're very excited to announce support of HTTPRoute for [routing inside the Mesh](https://kuma.io/docs/2.3.x/explore/gateway-api/#service-to-service-routing).

Reviewer: ```suggestion
In 2.3.0 we're excited to announce support of HTTPRoute for [routing inside the Mesh](https://kuma.io/docs/2.3.x/explore/gateway-api/#service-to-service-routing).
```

> It's a simple feature but excited to see what people build around it!

Reviewer: ```suggestion
It's a simple feature but we are excited to see what people build around it!
```

### Governance changes

The governance had not change since Kuma incubated in the CNCF.
This meant that it wasn't really a mature and thought through process.

> We've now spent significant time thinking about our [GOVERNANCE](https://github.com/kumahq/kuma/blob/master/GOVERNANCE.md).

Reviewer: ```suggestion
We've now spent significant time rethinking our [GOVERNANCE](https://github.com/kumahq/kuma/blob/master/GOVERNANCE.md).
```

> We believe this will make how the Kuma community works more transparent and more welcoming to new contributors.

Reviewer: ```suggestion
We believe it will make how the Kuma community works more transparent and welcoming to new contributors.
```

> By the way we are looking for production user stories, so if you have a good one please find us on Slack!

Reviewer: ```suggestion
By the way, we are looking for production user stories, so if you have a good one please find us on Slack!
```

## Upgrading

Be sure to carefully read the [Upgrade Guide](https://github.com/kumahq/kuma/blob/master/UPGRADE.md) before upgrading Kuma.

## Join the community!

> Join us on our [community channels](https://kuma.io/community/), including our official Slack chat, to learn more about Kuma.

Reviewer: ```suggestion
Join us on our [community channels](https://kuma.io/community/), including official Slack chat, to learn more about Kuma.
```
